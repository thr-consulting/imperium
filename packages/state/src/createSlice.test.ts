/* eslint-disable no-param-reassign */
import type {PayloadAction} from '@reduxjs/toolkit';
import {createSlice} from '@reduxjs/toolkit';

describe('create a slice', () => {
	it('should create a slice', () => {
		const slice = createSlice({
			name: 'name',
			initialState: {
				a: 5,
			},
			reducers: {
				setA: (st, action: PayloadAction<number>) => {
					st.a = action.payload;
				},
				whateverA: (st, action: PayloadAction<{setA: number}>) => {
					st.a = action.payload.setA;
				},
			},
		});

		expect(slice.name).toBe('name');
		const action = slice.actions.setA(2);
		expect(action).toMatchSnapshot();
		expect(slice.reducer({a: 5}, action)).toMatchSnapshot();
	});

	it('should create a slice with prepare', () => {
		const slice = createSlice({
			name: 'name',
			initialState: {
				a: 5,
			},
			reducers: {
				setA: {
					reducer: (st, action: PayloadAction<number>) => {
						st.a = action.payload;
					},
					prepare: (text: string) => {
						return {
							payload: parseInt(text, 10),
						};
					},
				},
			},
		});

		expect(slice.name).toBe('name');
		const action = slice.actions.setA('2');
		expect(action).toMatchSnapshot();
		expect(slice.reducer({a: 5}, action)).toMatchSnapshot();
	});
});
