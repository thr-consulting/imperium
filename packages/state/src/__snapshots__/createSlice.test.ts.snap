// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create a slice should create a slice 1`] = `
Object {
  "payload": 2,
  "type": "name/setA",
}
`;

exports[`create a slice should create a slice 2`] = `
Object {
  "a": 2,
}
`;

exports[`create a slice should create a slice with prepare 1`] = `
Object {
  "payload": 2,
  "type": "name/setA",
}
`;

exports[`create a slice should create a slice with prepare 2`] = `
Object {
  "a": 2,
}
`;
